src/Docker.DotNet/Endpoints/SwarmOperations.cs

```diff
@@ -13,10 +13,9 @@ internal class SwarmOperations : ISwarmOperations
     {
         internal static readonly ApiResponseErrorHandlingDelegate SwarmResponseHandler = (statusCode, responseBody) =>
         {
-            if (statusCode == HttpStatusCode.ServiceUnavailable)
+            if (statusCode < HttpStatusCode.OK || statusCode >= HttpStatusCode.BadRequest)

```

Comment:
This is what the default handler does. Should we just remove the custom handler?

Alternatively if you want to have a more 'typed' style exception you could have this read:

```C#
if (statusCode == HttpStatusCode.ServiceUnavailable)
{
    throw new DockerApiException(statusCode, "Node is not part of a swarm: " + responseBody);
}
else if (statusCode == HttpStatusCode.NotAcceptable)
{
    throw new DockerApiException(statusCode, "Node is already part of a swarm: " + responseBody);
}
```

Reply:
Or maybe just split it into a Positive/Negative match and update the functions as the API makes sense. Not sure if `ServiceUnavailable` or `NotAccepted` are ever returned from the functions for different meaning. Would need to look at the docs.

Reply:
I added a new exeption, dockerswarmexception which takes the deserialized message when created. So it's more typed style now, and I think it makes more sence now to have custom handler.

Reply:
The http status codes returned from docker api have different meaning, and a json is returned from docker api with a message, like "node is not part of a swarm".